File: HRMS.Core/Models/Event.cs
using HRMS.Core.Models.Enums;

namespace HRMS.Core.Models;

public class Event : Base
{
    public string Name { get; set; } = string.Empty;

    public EventType EventType { get; set; } = new();

    public DateTime StartDate { get; set; }

    public DateTime EndDate { get; set; }

    public string Description { get; set; } = string.Empty;

    public Employee Employee { get; set; } = new();

    public EventStatus Status { get; set; }
}

File: HRMS.Desktop/lib/models/event.dart
import 'package:json_annotation/json_annotation.dart';

import 'employee.dart';
import 'enums/event_status.dart';
import 'event_type.dart';

part 'event.g.dart';

@JsonSerializable()
class Event {
  int id;
  String name;
  String description;
  DateTime startDate;
  DateTime endDate;
  EventType? eventType;
  Employee? employee;
  EventStatus status;

  Event(
    this.id,
    this.name,
    this.description,
    this.startDate,
    this.endDate,
    this.eventType,
    this.employee,
    this.status
  );

  factory Event.fromJson(Map<String, dynamic> json) => _$EventFromJson(json);

  Map<String, dynamic> toJson() => _$EventToJson(this);
}
